docs/gitops-dashboard.md

```diff
@@ -27,5 +27,100 @@ Applications being managed by Weave GitOps are displayed in a list. Clicking the
+
+| Parameter                     | Type | Description                                                           | Default   |
+| ----------------------------- | - |--------------------------------------------------------------------- | --------- |
+| `--oidc-issuer-url`           | string | The URL of the issuer, typically the discovery URL without a path     |           |
+| `--oidc-issuer-client-id`     | string | The client ID that has been setup for Weave GitOps in the issuer      |           |
+| `--oidc-issuer-client-secret` | string | The client secret that has been setup for Weave GitOps in the issuer  |           |
+| `--oidc-redirect-url`         | string | The redirect URL that has been setup for Weave GitOps in the issuer, typically the dashboard URL followed by `/oauth2/callback ` |         |
+| `--oidc-cookie-duration`      | duration | The time duration that the ID Token HTTP cookie will remain valid, after successful authentication | "1h0m0s" |
+
+Ensure that your OIDC provider has been setup with a client ID/secret and the redirect URL of the dashboard.
+
+Once the HTTP server starts, it will redirect unauthenticated users to the provider's login page to authenticate them. Upon successful authentication, the users' identity will be impersonated in any calls made to the Kubernetes API, as part of any action they take in the dashboard. At this point, the dashboard will fail to render correctly unless RBAC has been configured accordingly. The following manifests represent the minimal set of permissions needed to view applications, commits and profiles from the dashboard:
+
+```yaml title="apps-reader.yaml"
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: apps-reader
+  namespace: wego-system
+rules:
+- apiGroups: ["wego.weave.works"]
+  resources: ["apps"]
+  verbs: [ "get", "list"]
+- apiGroups: ["source.toolkit.fluxcd.io"]
+  resources: ["gitrepositories"]
+  verbs: ["get"]
+- apiGroups: ["source.toolkit.fluxcd.io"]
+  resources: ["helmrepositories"]
+  verbs: ["get"]
+- apiGroups: ["kustomize.toolkit.fluxcd.io"]
+  resources: ["kustomizations"]
+  verbs: ["get"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get"]
+  resourceNames: ["wego-github-dev-cluster"] # name of secret created by Weave GitOps that contains the deploy key for the git repository
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: profiles-reader
+  namespace: default
+rules:
+- apiGroups: ["source.toolkit.fluxcd.io"]
+  resources: ["helmrepositories"]
+  verbs: ["get"]
+  resourceNames: [ "weaveworks-charts"]
+```
+
+The following manifest represents the minimal set of permissions needed to add applications from the dashboard:
+
+```yaml title="apps-writer.yaml"
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: apps-writer
+rules:
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get"]
```

Comment:
Is this the complete list of verbs needed for the writer?

Reply:
Yes, because everything else is handled via flux